### DISCUSSION TITLE
Automatically pause and resume a video

### DISCUSSION THREAD

**Discussion by yoniren2** [bonsai-rx/bonsai#1348] [Q&A]
Hello everyone, I am a new user who is only begin to undertsand the power of Bonsai and manage to make fast progress thanks to previous posts on this forum.
However, I am stuck now at trying to make a video that pauses automaticaly everytime an event is detected in the image, and resume once a key is pressed. Has someone the solution?

**Comment by yoniren2** [comment]
The idea is to wait fo the user to check if the detected event is real

**Comment by camille-lab** [comment]
Hey,
What kind of event are you trying to detect? 
To resume based on a key press, use the 'keydown' node

**Comment by yoniren2** [comment]
Hello camille :) 
Thank you for your reply

The video displays a fish in a tank and I need the video to pause when that fish enters a zone. It should stay on pause until I manually resume it (keydown).

**Comment by bruno-f-cruz** [comment]
You can toggle the `Playing` state of the `FileCapture` source:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/2e1eb5b3-bba4-4d7b-97e3-d2e058ac6516)




<details>
    <summary>Workflow (Copy+Paste me)</summary>
    
```xml


<?xml version="1.0" encoding="utf-8"?>
<WorkflowBuilder Version="2.7.3"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                 xmlns:rx="clr-namespace:Bonsai.Reactive;assembly=Bonsai.Core"
                 xmlns:cv="clr-namespace:Bonsai.Vision;assembly=Bonsai.Vision"
                 xmlns:wie="clr-namespace:Bonsai.Windows.Input;assembly=Bonsai.Windows.Input"
                 xmlns="https://bonsai-rx.org/2018/workflow">
  <Workflow>
    <Nodes>
      <Expression xsi:type="SubscribeSubject">
        <Name>PlayinigState</Name>
      </Expression>
      <Expression xsi:type="PropertyMapping">
        <PropertyMappings>
          <Property Name="Playing" />
        </PropertyMappings>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="cv:FileCapture">
          <cv:FileName />
          <cv:PlaybackRate>0</cv:PlaybackRate>
          <cv:StartPosition>0</cv:StartPosition>
          <cv:PositionUnits>Frames</cv:PositionUnits>
          <cv:Loop>false</cv:Loop>
          <cv:Playing>true</cv:Playing>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:DistinctUntilChanged" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="cv:Grayscale" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="wie:KeyDown">
          <wie:Filter>Space</wie:Filter>
          <wie:SuppressRepetitions>false</wie:SuppressRepetitions>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="IntProperty">
          <Value>1</Value>
        </Combinator>
      </Expression>
      <Expression xsi:type="rx:Accumulate" />
      <Expression xsi:type="Mod">
        <Operand xsi:type="IntProperty">
          <Value>2</Value>
        </Operand>
      </Expression>
      <Expression xsi:type="Equal">
        <Operand xsi:type="IntProperty">
          <Value>1</Value>
        </Operand>
      </Expression>
      <Expression xsi:type="rx:BehaviorSubject">
        <Name>PlayinigState</Name>
      </Expression>
    </Nodes>
    <Edges>
      <Edge From="0" To="1" Label="Source1" />
      <Edge From="1" To="2" Label="Source1" />
      <Edge From="2" To="3" Label="Source1" />
      <Edge From="3" To="4" Label="Source1" />
      <Edge From="5" To="6" Label="Source1" />
      <Edge From="6" To="7" Label="Source1" />
      <Edge From="7" To="8" Label="Source1" />
      <Edge From="8" To="9" Label="Source1" />
      <Edge From="9" To="10" Label="Source1" />
    </Edges>
  </Workflow>
</WorkflowBuilder>

```

</details>

**Reply by addy9908** [reply]
Hi bruno,
Is it possible to keep the default of playing as False? Every time I run a similar code again, it remembers previous state, rather than keep False as default. 

Thanks,
Zengyou

> You can toggle the `Playing` state of the `FileCapture` source:
> 
> ![image](https://private-user-images.githubusercontent.com/7049351/237396904-2e1eb5b3-bba4-4d7b-97e3-d2e058ac6516.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NzU4NDMsIm5iZiI6MTc0MjU3NTU0MywicGF0aCI6Ii83MDQ5MzUxLzIzNzM5NjkwNC0yZTFlYjViMy1iYmE0LTRkN2ItOTdlMy1kMmUwNThhYzY1MTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjFUMTY0NTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTk1ZjFkYjEyOTJjMTMwMWJmZDUyZGZkNTMwNjUyN2Q2ODdmYTA1OGZlNmE1MTEzNWI3NmQ4ZmJiMTE5YzllNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.intib9iJz7tDYxUUl7xeC3-x45HPBtCAU7xTIfh_okM)
> 
> Workflow (Copy+Paste me)

**Reply by bruno-f-cruz** [reply]
![image](https://github.com/user-attachments/assets/9424a661-10e1-4cfe-8f16-296e018b1b6b)
Initialize the subject elsewhere and use multicast from your keydown

**Reply by addy9908** [reply]
Thanks Bruno.
Another question: how can I use keydown or button to play the video frame by frame (keydown for next frame)? I want to score a video and give each frame a event name when I keydown, then move to next frame to keep scoring.
